Planner/app/src/main/java/rdatu/android/cyscorpions/com/projectplanner/model/TasksDatabaseHelper.java

```diff
@@ -0,0 +1,137 @@
+    private static String COLUMN_TASK_NAME = "task_name";
+    private static String COLUMN_TASK_DESC = "task_description";
+    private static String COLUMN_PLACE = "task_place";
+    private static String COLUMN_DATE = "task_date";
+    private static String COLUMN_TIMESLOT = "task_time";
+    private static String COLUMN_PRIORITY = "task_priority";
+    private static SQLiteDatabase mDatabase;
+
+    public TasksDatabaseHelper(Context context) {
+        super(context, DATABASE_NAME, null, 1);
+    }
+
+
+    @Override
+    public void onCreate(SQLiteDatabase db) {
+        db.execSQL("create table tasks (task_date varchar(30),task_time varchar(15), task_name varchar(100), task_description varchar(200), task_place varchar(100),task_priority varchar(10))");
+        mDatabase = db;
+    }
+
+    public long insertTasks(String name, String desc, String place, String date, String time, String priority) {
+        ContentValues cv = new ContentValues();
+        cv.put(COLUMN_DATE, date);
+        cv.put(COLUMN_TIMESLOT, time);
+        cv.put(COLUMN_TASK_NAME, name);
+        cv.put(COLUMN_TASK_DESC, desc);
+        cv.put(COLUMN_PLACE, place);
+        cv.put(COLUMN_PRIORITY, priority);
+        return getWritableDatabase().insert(TABLE_TASKS, null, cv);
+    }
+
+    public void deleteAll() {
+        getWritableDatabase().delete(TABLE_TASKS, null, null);
+    }
+
+    public void deleteWithCondition(String date, String time) {
+        String Where = COLUMN_DATE + "='" + date + "' AND " + COLUMN_TIMESLOT + "='" + time + "'";
+        try {
+            getWritableDatabase().execSQL("DELETE FROM tasks WHERE " + Where);
+        } catch (Exception e) {
+            Log.e("Planner", "Error:", e);
+        }
+
+    }
+
+    public void updateTask(String name, String descr, String time, String date, String place, String priority) {
+        try {
+            getWritableDatabase().execSQL("UPDATE tasks SET task_name=?, task_description=?,task_place=?,task_priority=? WHERE task_date=? AND task_time =?", new String[]{name, descr, place, priority, date, time});
+        } catch (Exception e) {
+            Log.e("Planner", "Error: ", e);
+        }
+    }
+
+
+    public TaskCursor queryTaskForDate(String date) {
+        Cursor wrapped = getReadableDatabase().rawQuery("SELECT * FROM tasks WHERE task_date= '" + date + "'", null);
+        return new TaskCursor(wrapped);
+    }
+
+    public TaskCursor queryTask() {
+        Cursor wrapped = getReadableDatabase().query(TABLE_TASKS, null, null, null, null, null, null);
+        return new TaskCursor(wrapped);
+    }
+
+    public TaskCursor queryTask(String date, String time) {
+        Cursor wrapped = getReadableDatabase().rawQuery("SELECT * FROM tasks WHERE task_date ='" + date + "' AND task_time ='" + time + "'", null);
+        return new TaskCursor(wrapped);
+    }
+
+    @Override
+    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
```

Comment:
About the onUpgrade method po. It is required to implement the method po, even though it is blank. will return an error otherwise po.